### DISCUSSION TITLE
EnergyOpt

### DISCUSSION THREAD

**Discussion by jbakosi** [UIUC-PPL/charm#3694] [General] [answered]
Hi folks,

Is [EnergyOpt](http://charm.cs.illinois.edu/newPapers/19-05/Charm___entry_dvfs_paper.pdf) available in a latest release and/or main? I tried grepping for 'energyopt' but get no hits. If so, is it enabled differently?

Also, if available, do I need some non-user permissions on a HPC machine to be able to use DVFS?

Thanks,
J

**Comment by kavithachandrasekar** [comment, marked as answer]
The energy optimization code is in a branch `bilge/energy_opt` and hasn’t been updated for a while, we can look to update it to merge it into main. The feature can be enabled by using `--enable-energyopt` build option and it does require non-user privileges since the energy optimization code in charm++ runtime makes changes to CPU frequency using DVFS.

Thanks,
Kavitha

**Comment by jbakosi** [comment]
Thank you, Kavita,

I would like to try the energyopt functionality. If it is not much trouble, could you please merge it into main?

Thank you,
Jozsef
